presto-main/src/main/java/com/facebook/presto/operator/GroupByHash.java

```diff
@@ -64,4 +64,6 @@ static GroupByHash createGroupByHash(
     boolean contains(int position, Page page, int[] hashChannels);
 
     int putIfAbsent(int position, Page page);
+
+    int compare(int leftGroupId, int rightGroupId);
```

Comment:
That's a good idea. Changed to `compare` to `getRawHash`.

There are only two drawbacks. First is that result of such `MergeHashSort` is not ordered strictly speaking because of hash collisions. Rows with same hash value can interleaved. It makes it a little be hacky from this perspective, but doesn't change correctness (as long as positions with same hash are merged into one single `Page` - which is happening).

Second drawback is that it comparing is faster then calculating hash for multiple columns - for hash we always need to read/touch all values, but that's shouldn't be that big a deal.